lib/src/execute.rs

```diff
@@ -406,14 +435,54 @@ impl ExecuteCtx {
             compliance_region: Vec::from(REGION_NONE),
         };
 
-        let res = self.reuse_or_spawn_guest(req, metadata).await;
+        let (resp, mut err) = self.reuse_or_spawn_guest(req, metadata).await;
 
         let span = info_span!("request", id = req_id);
         let _span = span.enter();
 
-        info!("response status: {:?}", res.0.status());
+        info!("response status: {:?}", resp.status());
+
+        if let Some(e) = err {
+            match e.downcast::<NonHttpResponse>() {
+                Ok(NonHttpResponse::PushpinRedirect(redirect_info)) => {
+                    let backend_name = redirect_info.backend_name;
+                    let redirect_request_info = redirect_info.request_info;
+                    info!("Pushpin redirect signaled to backend '{}'", backend_name);
+
+                    let local_pushpin_proxy_port = match local_pushpin_proxy_port {
```

Comment:
I don't think you need the clone at the start of the method. Since it should be `Copy`, I believe you can just make this:

```suggestion
                    let local_pushpin_proxy_port = match self.local_pushpin_proxy_port {
```